builder/vmware/iso/step_create_vmx.go

```diff
@@ -155,7 +156,34 @@ func (s *stepCreateVMX) Run(state multistep.StateBag) multistep.StepAction {
 	return multistep.ActionContinue
 }
 
-func (s *stepCreateVMX) Cleanup(multistep.StateBag) {
+func (s *stepCreateVMX) Cleanup(state multistep.StateBag) {
+	driver := state.Get("driver").(vmwcommon.Driver)
+	config := state.Get("config").(*Config)
+	ui := state.Get("ui").(packer.Ui)
+
+	ui.Say("Unregistering and deleting virtual machine...")
+	var err error = nil
+	for i := 0; i < 5; i++ {
+		running, err := driver.IsRunning(config.VMXTemplatePath)
+		if err == nil {
+			if running {
+				err = driver.Stop(config.VMXTemplatePath)
+			}
+		}
+		if err == nil {
```

Comment:
It looks like this block should be merged with the preceding block since the conditional check is the same. If you're intending to check the error from `err = driver.Stop(config.VMXTemplatePath)` the flow is not super obvious.